google-cloud-spanner/lib/google/cloud/spanner/client.rb

```diff
@@ -1092,16 +1228,30 @@ def delete table, keys = [], call_options: nil
         #     c.insert "users", [{ id: 2, name: "Harvey",  active: true }]
         #   end
         #
-        def commit call_options: nil, &block
+        # @example Commit using tag for transaction statistics collection.
+        #   require "google/cloud/spanner"
+        #
+        #   spanner = Google::Cloud::Spanner.new
+        #
+        #   db = spanner.client "my-instance", "my-database"
+        #
+        #   db.commit tag: "Tag-1" do |c|
+        #     c.update "users", [{ id: 1, name: "Charlie", active: false }]
+        #     c.insert "users", [{ id: 2, name: "Harvey",  active: true }]
+        #   end
+        def commit tag: nil, call_options: nil, &block
```

Comment:
The commit (transaction) `tag` should be defined in the transaction creation, like in your previous example:

```
db.transaction tag: "Tag-1" do |tx| ...
```

The above transaction tag should be included in the commit statement.

If this is for hand-crafted transactions, I still think we should define the transaction tag when the user creates the transaction initially.

Reply:
Or is this method used internally only?

Reply:
`commit` is public method. 

For hand-crafted transaction :
https://github.com/googleapis/google-cloud-ruby/blob/880f64fe6db2d48ef989fad356a21d23550cbd88/google-cloud-spanner/lib/google/cloud/spanner/client.rb#L1352-L1373

In this 'Tag-1' set to a transaction(L1364) which is use to set a transaction tag in API query calls executed on that transaction.

Reply:
Got it, so do we need the tag in the `commit` method? I think we should remove that, since it should be set during the transaction creation, right? Otherwise, what would happen if the user sets a transaction tag in the creation and another one in the commit?

Reply:
I think no need to remove it. There is a commit method in the client so the user might directly call it, while Transaction calls the internal private service object commit method. 

Client : user can directly use.
https://github.com/googleapis/google-cloud-ruby/blob/bf17920882c3297f66d2249cd023bb2883e707b6/google-cloud-spanner/lib/google/cloud/spanner/client.rb#L1095

Service: Private 
https://github.com/googleapis/google-cloud-ruby/blob/bf17920882c3297f66d2249cd023bb2883e707b6/google-cloud-spanner/lib/google/cloud/spanner/service.rb#L399